src/parser/utils.rs

```diff
@@ -23,10 +23,43 @@ use crate::{
     types::{TypeF, Types},
 };
 
-/// Distinguish between the standard string separators `"`/`"` and the multi-line string separators
-/// `m%"`/`"%` in the parser.
+/// Distinguish between the standard string opening delimiter `"`, the multi-line string
+/// opening delimter `m%"`, and the symbolic string opening delimiter `s%"`.
 #[derive(Copy, Clone, Eq, PartialEq, Debug)]
-pub enum StringKind {
+pub enum StringStartDelimiter {
+    Standard,
+    Multiline,
+    Symbolic,
+}
+
+impl StringStartDelimiter {
+    pub fn is_closed_by(&self, close: &StringEndDelimiter) -> bool {
+        matches!(
+            (self, close),
+            (StringStartDelimiter::Standard, StringEndDelimiter::Standard)
+                | (
+                    StringStartDelimiter::Multiline,
+                    StringEndDelimiter::Multiline
```

Comment:
Nitpick: should we call this `Multiline` if it also works for `Symbolic`? It looks inconsistent with `StringStartDelimiter`. Maybe `Special`, if we want to call all those `xxx%"` strings "special strings".

Reply:
I'm not so sure about "special" as a term of art here. I'm not sure it really means anything beyond "these are different somehow".

What about just being very literal and using `StringEndDelimiter::Quote` and `StringEndDelimiter::QuotePercent`?

Reply:
> I'm not so sure about "special" as a term of art here. I'm not sure it really means anything beyond "these are different somehow".

Yes.... in this case I can't think of something better than just saying "it's not the normal string delimiter".

> What about just being very literal and using StringEndDelimiter::Quote and StringEndDelimiter::QuotePercent?

It's a possible solution as well. It has the advantage of being descriptive. Though it's tied to the syntax instead of the semantics, so if we change `%` for `#`, we'll have to change that.

Both are reasonable!

Reply:
Fair point about coupling the syntax and semantics. I'll go with `Special` for now because I can't think of anything better. :sweat_smile: